City-of-Helsinki/berth-reservations-admin
# VEN-618 | Store the new user in localStorage before trying to access the app

## Description :sparkles:
- Await the new tokens to be fetched and the new user to be stored in `localStorage`
- Clear `localStorage` when the user logs out
- Writing unit tests

## Issues :bug:

### Closes :no_good_woman:
[VEN-618](https://helsinkisolutionoffice.atlassian.net/browse/VEN-618): Login requires manual refresh or a second attempt to access the application 

### Related :handshake:
This also should fix the [error in fetching the data in the initial request](https://helsinkicity.slack.com/archives/CBA1QHTHN/p1589272220364600)

## Testing :alembic:

### Automated tests :gear:️
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/1ff303da090a47b40419c9712a24b0721973f278/src/domain/auth/authService.test.js#L161
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/1ff303da090a47b40419c9712a24b0721973f278/src/domain/auth/authService.test.js#L215

### Manual testing :construction_worker_man:
- Log out
- Log in
- The user should be redirected back to the front page (harbors) instead of the application's login screen
